comp/logs/agent/config/endpoints.go

```diff
@@ -56,9 +58,142 @@ type Endpoint struct {
+	UseSSL     *bool  `mapstructure:"use_ssl" json:"use_ssl"`
+
+	Endpoint `mapstructure:",squash"`
+}
+
+// NewEndpoint returns a new Endpoint with the minimal field initialized.
+func NewEndpoint(apiKey string, host string, port int, useSSL bool) Endpoint {
+	apiKey = pkgconfigutils.SanitizeAPIKey(apiKey)
+	return Endpoint{
+		apiKeyGetter: func() string { return apiKey },
+		Host:         host,
+		Port:         port,
+		useSSL:       useSSL,
+		isReliable:   true, // by default endpoints are reliable
+	}
+}
+
+// NewTCPEndpoint returns a new TCP  Endpoint based on LogsConfigKeys
+func NewTCPEndpoint(logsConfig *LogsConfigKeys) Endpoint {
+	return Endpoint{
+		apiKeyGetter:            logsConfig.getAPIKeyGetter(),
+		ProxyAddress:            logsConfig.socks5ProxyAddress(),
+		ConnectionResetInterval: logsConfig.connectionResetInterval(),
+		useSSL:                  logsConfig.logsNoSSL(),
+		isReliable:              true, // by default endpoints are reliable
+	}
+}
+
+// NewHTTPEndpoint returns a new HTTP  Endpoint based on LogsConfigKeys
+func NewHTTPEndpoint(logsConfig *LogsConfigKeys) Endpoint {
+	return Endpoint{
+		apiKeyGetter:            logsConfig.getAPIKeyGetter(),
+		UseCompression:          logsConfig.useCompression(),
+		CompressionLevel:        logsConfig.compressionLevel(),
+		ConnectionResetInterval: logsConfig.connectionResetInterval(),
+		BackoffBase:             logsConfig.senderBackoffBase(),
+		BackoffMax:              logsConfig.senderBackoffMax(),
+		BackoffFactor:           logsConfig.senderBackoffFactor(),
+		RecoveryInterval:        logsConfig.senderRecoveryInterval(),
+		RecoveryReset:           logsConfig.senderRecoveryReset(),
+		useSSL:                  logsConfig.logsNoSSL(),
+		isReliable:              true, // by default endpoints are reliable
+	}
+}
+
+// The setting from 'logs_config.additional_endpoints' is directly unmarshalled from the configuration into a
+// []unmarshalEndpoint and do not use the constructors. In this case, apiKeyGetter is initialized to returned the API
+// key from the loaded data instead of 'api_key'/'logs_config.api_key'.
+
+func loadTCPAdditionalEndpoints(main Endpoint, l *LogsConfigKeys) []Endpoint {
+	additionals := l.getAdditionalEndpoints()
+
+	newEndpoints := make([]Endpoint, 0, len(additionals))
+	for _, e := range additionals {
+		newE := NewEndpoint(e.APIKey, e.Host, e.Port, false)
+
+		newE.UseCompression = e.UseCompression
+		newE.CompressionLevel = e.CompressionLevel
+		newE.ProxyAddress = l.socks5ProxyAddress()
+		newE.isReliable = e.IsReliable == nil || *e.IsReliable
+		newE.ConnectionResetInterval = e.ConnectionResetInterval
+		newE.BackoffFactor = e.BackoffFactor
+		newE.BackoffBase = e.BackoffBase
+		newE.BackoffMax = e.BackoffMax
+		newE.RecoveryInterval = e.RecoveryInterval
+		newE.RecoveryReset = e.RecoveryReset
+		newE.Version = e.Version
+		newE.TrackType = e.TrackType
+		newE.Protocol = e.Protocol
+		newE.Origin = e.Origin
+
+		if e.UseSSL == nil {
```

Comment:
nitpick, reverse this conditional:
```suggestion
		if e.UseSSL != nil {
			newE.useSSL = *e.UseSSL
		} else {
			newE.useSSL = main.useSSL
		}
```
It keeps the repetition of `e.UseSSL` closer together and makes it more obvious that `main.useSSL` is the backend plan.